scripts/imgtool/main.py

```diff
@@ -477,10 +487,46 @@ def sign(key, public_key_format, align, version, pad_sig, header_size,
         }
 
     img.create(key, public_key_format, enckey, dependencies, boot_record,
-               custom_tlvs, int(encrypt_keylen), clear, baked_signature,
+               custom_tlvs, compression_tlvs, int(encrypt_keylen), clear, baked_signature,
                pub_key, vector_to_sign)
-    img.save(outfile, hex_addr)
 
+    if re.match('.*\.bin$', infile) is not None :
+        if compression == "lzma2" :
+            compressed_img = image.Image(version=decode_version(version), header_size=header_size,
+                      pad_header=pad_header, pad=pad, confirm=confirm,
+                      align=int(align), slot_size=slot_size,
+                      max_sectors=max_sectors, overwrite_only=overwrite_only,
+                      endian=endian, load_addr=load_addr, rom_fixed=rom_fixed,
+                      erased_val=erased_val, save_enctlv=save_enctlv,
+                      security_counter=security_counter, max_align=max_align)
+            compressed_file =  re.sub('zephyr\.bin$','zephyr.compressed.bin',infile)
```

Comment:
You might use `tempfile.TemporaryFile` to store the compressed image not to pollute the build directory. Also, I don't think you should assume `zephyr.bin` file name here.